FreeRTOS/FreeRTOS-Plus-TCP
# Add build combination checks

Description
-----------
This PR adds build combination checks to the FreeRTOS-Plus-TCP repository to check the 'buildability' of the source code when various options are switched on or off in FreeRTOSIPConfig.h.
The checks included in this PR are just of two kinds: One which enables everything and one which disables everything.

Test Steps
-----------
- Go to the `test/build-combination` directory.
- Run `cmake -S . -B build -DTEST_CONFIGURATION=DISABLE_ALL` **OR**
                 `cmake -S . -B build -DTEST_CONFIGURATION=ENABLE_ALL`.
- Go to build directory: `cd build`.
- Run `make`.

The make command should finish successfully.

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
